engine/resources/exec_test.go

```diff
@@ -257,3 +260,62 @@ func TestExecTimeoutBehaviour(t *testing.T) {
 
 	// no error
 }
+
+func TestExecAutoEdge1(t *testing.T) {
+
+	g, err := pgraph.NewGraph("TestGraph")
+	if err != nil {
+		t.Errorf("error creating graph: %v", err)
+		return
+	}
+
+	r1 := &ExecRes{
```

Comment:
Can you build r2, r3, and r4 with `NewNamedResource` instead? You'll also have to convert it to the *ExecRes afterwards to be able to set these additional properties, but I think that will be cleaner if it works.

Reply:
Done